creation/web_base/glidein_startup.sh

```diff
@@ -152,1932 +76,533 @@ do_start_all() {
-    esac
-}
-
-#####################
-# Check signature
-check_file_signature() {
-    cfs_id="$1"
-    cfs_fname="$2"
-
-    cfs_work_dir="$(get_work_dir "${cfs_id}")"
-
-    cfs_desc_fname="${cfs_work_dir}/${cfs_fname}"
-    cfs_signature="${cfs_work_dir}/signature.sha1"
-
-    if [ "${check_signature}" -gt 0 ]; then # check_signature is global for simplicity
-        tmp_signname="${cfs_signature}_$$_$(date +%s)_${RANDOM}"
-        if ! grep " ${cfs_fname}$" "${cfs_signature}" > "${tmp_signname}"; then
-            rm -f "${tmp_signname}"
-            echo "No signature for ${cfs_desc_fname}." 1>&2
+    for tk in "$(pwd)/credential_"*".idtoken"; do
+        echo "Setting GLIDEIN_CONDOR_TOKEN to ${tk} " 1>&2
+        num_gct=$(( num_gct + 1 ))
+        export GLIDEIN_CONDOR_TOKEN="${tk}"
+        fullpath="$(readlink -f "${tk}" )"
+        if [ $? -eq 0 ]; then
+            echo "Setting GLIDEIN_CONDOR_TOKEN ${tk} to canonical path ${fullpath}" 1>&2
+            export GLIDEIN_CONDOR_TOKEN="${fullpath}"
         else
-            (cd "${cfs_work_dir}" && sha1sum -c "${tmp_signname}") 1>&2
-            cfs_rc=$?
-            if [ ${cfs_rc} -ne 0 ]; then
-                "${main_dir}"/error_augment.sh -init
-                "${main_dir}"/error_gen.sh -error "check_file_signature" "Corruption" "File $cfs_desc_fname is corrupted." "file" "${cfs_desc_fname}" "source_type" "${cfs_id}"
-                "${main_dir}"/error_augment.sh  -process ${cfs_rc} "check_file_signature" "${PWD}" "sha1sum -c ${tmp_signname}" "$(date +%s)" "(date +%s)"
-                "${main_dir}"/error_augment.sh -concat
-                warn "File ${cfs_desc_fname} is corrupted."
-                rm -f "${tmp_signname}"
-                return 1
-            fi
-            rm -f "${tmp_signname}"
-            echo "Signature OK for ${cfs_id}:${cfs_fname}." 1>&2
+            echo "Unable to get canonical path for GLIDEIN_CONDOR_TOKEN ${tk}" 1>&2
         fi
-    fi
-    return 0
-}
-
-#####################
-# Untar support func
-
-get_untar_subdir() {
-    gus_id="$1"
-    gus_fname="$2"
-
-    gus_prefix="$(get_prefix "${gus_id}")"
-    gus_config_cfg="${gus_prefix}UNTAR_CFG_FILE"
+    done
+    if [ ! -f "${GLIDEIN_CONDOR_TOKEN}" ] ; then
+        token_err_msg="problem setting GLIDEIN_CONDOR_TOKEN"
+        token_err_msg="${token_err_msg} will attempt to recover, but condor IDTOKEN auth may fail"
+        echo "${token_err_msg}"
+        echo "${token_err_msg}" 1>&2
+    fi
+    if [ ! "${num_gct}" -eq  1 ] ; then
+        token_err_msg="WARNING  GLIDEIN_CONDOR_TOKEN set ${num_gct} times, should be 1 !"
+        token_err_msg="${token_err_msg} condor IDTOKEN auth may fail"
+        echo "${token_err_msg}"
+        echo "${token_err_msg}" 1>&2
+    fi
+
+    ########################################
+    prepare_workdir
+
+    ######################################
+    # extract and source all the data contained at the end of this script as tarball
+    extract_all_data
```

Comment:
This extracts and sources all the extracted files (so functions and variables are defined) 
Could be renamed `extract_and_source_all_data `